### PR TITLE
When no records match, reply with NXDOMAIN

### PR DESCRIPTION
**Pull request by chrisohaver**
Reply with NXDOMAIN, when there are no matching records.

Addresses issue #357.

### CODE REVIEW

File: middleware/kubernetes/handler.go [hunk 1]
```diff
@@ -87,7 +87,7 @@ func (k Kubernetes) ServeDNS(ctx context.Context, w dns.ResponseWriter, r *dns.M
 	}
 
 	if len(records) == 0 {
-		return middleware.BackendError(&k, zone, dns.RcodeSuccess, state, nil /*debug*/, nil, middleware.Options{})
+		return middleware.BackendError(&k, zone, dns.RcodeNameError, state, nil /*debug*/, nil, middleware.Options{})
```
Comment (miekg) [inline middleware/kubernetes/handler.go]: NameErrorr should be handled by the k.IsNameError() above, but k8s already returns `false` for that. I.e. that would be the proper fix.

**Review by miekg — CHANGES_REQUESTED** [review]

**Comment by chrisohaver** [comment]
Reflecting on this change last evening... I think I need to add differentiation between no records, and non-existent domain.

File: middleware/kubernetes/kubernetes.go [hunk 2]
```diff
@@ -57,7 +59,7 @@ func (k *Kubernetes) Lookup(state request.Request, name string, typ uint16) (*dn
 // IsNameError implements the ServiceBackend interface.
 // TODO(infoblox): implement!
 func (k *Kubernetes) IsNameError(err error) bool {
-	return false
+	return err == noItemsErr
```
Comment (miekg) [inline middleware/kubernetes/kubernetes.go:62]: Is this really the proper way k8s signals nothing found?

**Comment by miekg** [comment]
well, to be precise, NODATA and NXDOMAIN are two different replies for non-existence.

**Comment by chrisohaver** [comment]
Yeah - that's what i was referring to in my last comment.

**Pull request closed without merging**